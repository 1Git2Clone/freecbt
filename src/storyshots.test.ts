import initStoryshots from "@storybook/addon-storyshots";

// Various mocks for quieter logging during tests:
//
// * Quiet, StatusBar
jest.useFakeTimers();

// * Quiet, posed. Thanks, https://stackoverflow.com/a/59593847
jest.mock("react-native/Libraries/Animated/src/NativeAnimatedHelper");

// * Quiet, AsyncStorage. I'd love to upgrade you, but can't yet - we're locked
// to the version used by Expo.
const consoleError = console.error;
jest.spyOn(console, 'error').mockImplementation(msg => {
  if (msg !== "Warning: AsyncStorage has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-community/async-storage' instead of 'react-native'. See https://github.com/react-native-community/react-native-async-storage") {
    consoleError.call(console, msg);
  }
});

initStoryshots();
