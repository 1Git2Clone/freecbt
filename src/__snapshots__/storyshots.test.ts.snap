// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots DebugScreen default view 1`] = `
<RCTScrollView>
  <View>
    <Text
      style={
        Object {
          "borderBottomWidth": 1,
          "fontSize": 24,
        }
      }
    >
      Debug
    </Text>
    <View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Release channel
          : 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          (dev)
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Installation id
          : 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          a01650bb-918d-40be-87be-cf376ab6189f
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Expo version
          : 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          mock
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          App version
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Revision
          : 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          (dev)
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Revision Git
          : 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          2.2.0-51-g22c6ffd-dirty
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Revision Date
          : 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          Wed, 03 Jun 2020 04:20:59 -0400
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Revision Timestamp
          : 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          1591172459
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          OS
          : 
        </Text>
        <Text
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          ios
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          debugVisible
          : 
        </Text>
        <View>
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          reminders
          : 
        </Text>
        <View>
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          remindersEachMinute
          : 
        </Text>
        <View>
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text>
          Dump AsyncStorage?
          : 
        </Text>
        <View>
          <RCTSwitch
            accessibilityRole="button"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 31,
                "width": 51,
              }
            }
            value={false}
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Storyshots ExampleView default view 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello Storybook
  </Text>
</View>
`;
